include/oneapi/tbb/task_arena.h

```diff
@@ -95,6 +95,11 @@ TBB_EXPORT void __TBB_EXPORTED_FUNC isolate_within_arena(d1::delegate_base& d, s
 TBB_EXPORT void __TBB_EXPORTED_FUNC enqueue(d1::task&, d1::task_arena_base*);
 TBB_EXPORT void __TBB_EXPORTED_FUNC enqueue(d1::task&, d1::task_group_context&, d1::task_arena_base*);
 TBB_EXPORT void __TBB_EXPORTED_FUNC submit(d1::task&, d1::task_group_context&, arena*, std::uintptr_t);
+
+#if __TBB_PREVIEW_PARALLEL_BLOCK
+TBB_EXPORT void __TBB_EXPORTED_FUNC register_parallel_block(d1::task_arena_base&);
+TBB_EXPORT void __TBB_EXPORTED_FUNC unregister_parallel_block(d1::task_arena_base&, bool);
```

Comment:
Do we anticipate any future extension? In other words, does it make sense to accept an untyped pointer for any future details?

```suggestion
TBB_EXPORT void __TBB_EXPORTED_FUNC unregister_parallel_block(d1::task_arena_base&, void* /*supplemental details*/);
```

Also, I don't think single bit can be passed. At least a word (or even more?) is used anyway. Then, it makes sense to accept at least that width and do not kill opportunity for future extensions:
```suggestion
TBB_EXPORT void __TBB_EXPORTED_FUNC unregister_parallel_block(d1::task_arena_base&, unsigned short/*supplemental details*/);
```
Please double check what size of information is passed.

Reply:
Yeah, I also have been thinking about that. Perhaps we should pass `std::uintptr_t` instead of `void*`?

Reply:
Following the logic of getting ready for future extensions, perhaps the "register" function should also take an extra argument.